src/config.ts

```diff
@@ -138,18 +137,32 @@ function encodeStoredItem(data) {
  * @param {*} data 
  */
 function decodeStoredItem(id: string, data) {
-    if(typeof data !== "string") return data;
-    
-    try {
-        let str = JSON.parse(data);
-        
-        if(!Array.isArray(str)) return data;
-        return new SBMap(id, str);
-    } catch(e) {
-
-        // If all else fails, return the data
-        return data;
+    if (!Config.defaults[id]) return data;
+
+    if (Config.defaults[id] instanceof SBMap) {
+        try {
+            if (!Array.isArray(data)) return data;
+            return new SBMap(id, data);
+        } catch(e) {
+            console.error("Failed to parse SBMap: "+ id);
+        }
+    }
+
+    // This is the old format for SBMap (a JSON string)
+    if (typeof data === "string") {
```

Comment:
I was trying to prevent this.
I don't want a string being converted to a map.

Reply:
This is required to migrate the old data right?

Reply:
Well it should be migrated after the version change
not on every read however I don't think any map needs to be migrated

Reply:
It's easier to just migrate it this way. It can be removed in a few months.

Reply:
It would be better to have it during the update process.
at least it will be fixed some day :D